torch/_prims_common/__init__.py

```diff
@@ -30,8 +35,20 @@
 
     import sympy
 
-import torch
-from torch import sym_float, sym_int, sym_max
+    class _WorksWithInt(typing.Protocol):
+        def __add__(self, other: Any) -> typing.Self:
+            ...
+
+        def __radd__(self, other: Any) -> typing.Self:
+            ...
+
+        def __mul__(self, other: Any) -> typing.Self:
+            ...
+
+        def __rmul__(self, other: Any) -> typing.Self:
+            ...
+
+    _T = TypeVar("_T", bound=_WorksWithInt)
```

Comment:
nit: In general I don't like adding bounds to a generic `_T` like this (because usually `_T` would just be an unbound generic). I would prefer a name like `_WorksWithIntT` so the use sites are clearer.

Reply:
In this case the caller needs to preserve their types (either sympy.Expr when used in inductor, SymInt when used in primtorch), but the code in this file needs the protocal.  I couldn't get it to type cleanly without this.

Reply:
I'm just bikeshedding on the name `_T`.  When I see a raw `_T` my assumption is that it's unbound. Naming it something like `_WorksWithIntT` would (maybe) make it clear when reading the use sites that it's a generic but also make it clear that it's not an unbound generic.

To me:
```
def make_channels_last_1d_strides_for(
    shape: Sequence[_T],
) -> Tuple[Union[_T, int], ...]:
```
says this can take a Sequence of any type and returns a tuple with that same type (and int).

but
```
def make_channels_last_1d_strides_for(
    shape: Sequence[_WorksWithIntT],
) -> Tuple[Union[_WorksWithIntT, int], ...]:
```
says that this takes a Sequence of a generic type which is bound by _WorksWithInt and returns a tuple with that same type (and int).

Reply:
I'll rename it to that.